vscode/src/createProject.ts

```diff
@@ -265,93 +254,175 @@ export async function initProjectCreator(context: vscode.ExtensionContext) {
-
-        // Ask the user to pick a project to add as a reference
-        const projectChoice = await vscode.window.showQuickPick(
-          projectChoices.map((choice) => {
-            if ("github" in choice.ref) {
-              return {
-                label: choice.name,
-                detail: `github://${choice.ref.github.owner}/${choice.ref.github.repo}#${choice.ref.github.ref}`,
-                iconPath: githubIcon,
-                ref: choice.ref,
-              };
-            } else {
-              return {
-                label: choice.name,
-                detail: choice.ref.path,
-                iconPath: folderIcon,
-                ref: choice.ref,
-              };
-            }
-          }),
-          { placeHolder: "Pick a project to add as a reference" },
-        );
-
-        if (!projectChoice) {
-          log.info("User cancelled project choice");
+        if (!importChoice) {
+          log.info("User cancelled import choice");
           return;
         }
 
-        log.info("User picked project: ", projectChoice);
-
-        if (!manifestObj["dependencies"]) manifestObj["dependencies"] = {};
-        manifestObj["dependencies"][projectChoice.label] = projectChoice.ref;
-
-        // Apply the edits to the qsharp.json
-        const edit = new vscode.WorkspaceEdit();
-        edit.replace(
-          qsharpJsonUri,
-          new vscode.Range(0, 0, qsharpJsonDoc.lineCount, 0),
-          JSON.stringify(manifestObj, null, 2),
-        );
-        if (!(await vscode.workspace.applyEdit(edit))) {
-          vscode.window.showErrorMessage(
-            "Unable to update the qsharp.json file. Check the file is writable",
+        if (importChoice === "Import from GitHub") {
+          await importExternalPackage(
+            qsharpJsonDoc,
+            qsharpJsonUri,
+            manifestObj,
+          );
+        } else {
+          await importLocalPackage(
+            qsharpJsonDoc,
+            qsharpJsonDir,
+            qsharpJsonUri,
+            manifestObj,
           );
-          return;
         }
-
-        // Bring the qsharp.json to the front for the user to save
-        await vscode.window.showTextDocument(qsharpJsonDoc);
       },
     ),
   );
+  async function importLocalPackage(
+    qsharpJsonDoc: vscode.TextDocument,
+    qsharpJsonDir: vscode.Uri,
+    qsharpJsonUri: vscode.Uri,
+    manifestObj: any,
+  ) {
+    // Find all the other Q# projects in the workspace
+    const projectFiles = (
+      await vscode.workspace.findFiles("**/qsharp.json")
+    ).filter((file) => file.toString() !== qsharpJsonUri.toString());
+
+    const projectChoices: Array<{ name: string; ref: LocalProjectRef }> = [];
+
+    projectFiles.forEach((file) => {
+      const dirName = file.path.slice(0, -"/qsharp.json".length);
+      const relPath = getRelativeDirPath(qsharpJsonDir.path, dirName);
+      projectChoices.push({
+        name: dirName.slice(dirName.lastIndexOf("/") + 1),
```

Comment:
I can't find any bugs from just reading the code, but the `/` separator has me wondering if this has been tested on Windows (desktop and web)? LMK if you want me to give it a spin on my laptop.

-- in my experience, Windows browser, Windows desktop, and linux/mac browser/desktop all have unique path handling concerns. And even just in the browser: the playground, the test FS, and the local FS all show up as different virtual filesystems with different quirks. So I've become very distrustful of filesystem code... forgive me.

Reply:
FWIW, this is previously-existing code from Bill's implementation, I just moved it elsewhere. It's been live for a while now, and I believe all paths in the `vscode` APIs are normalized  -- I can't find the exact docs but this hints at it https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L1466

Reply:
I think Mine is right. In doing some testing on my Windows ARM VM I noticed the relative paths it generated go all the way to the root of the drive then back down, and this spot may well be the cause. Good catch!

Reply:
I've updated this to use the vscode URI API which normalizes paths, however I don't have a readily available windows machine to test. I've read the [source](https://github.com/microsoft/vscode/blob/main/src/vs/base/common/uri.ts) for this URI implementation, and I believe it does do the normalization we need.